include/onnxruntime/core/framework/execution_provider.h

```diff
@@ -259,6 +259,9 @@ class IExecutionProvider {
    */
   virtual int GenerateMetaDefId(const onnxruntime::GraphViewer& graph_viewer, uint64_t& model_hash) const;
 
```

Comment:
this should actually be non-const as it's changing the internal state of the model id generator (my mistake with the original code - the unique_ptr hides that misuse).

Reply:
const seems needed if GenerateModelId is used in GetCapability